### ISSUE TITLE
Q: Plans for a new release?

### ISSUE THREAD

**Issue by b-abel** [matcornic/hugo-theme-learn#502]
Hi there!
Are there any plans for when a new version will be released? The issue #378 is very striking and we would love to use the already existing fix. But for this, we would need a released version.

**Comment by muffmolch** [comment]
project seems dead ... more or less .. :-(

**Comment by mirisbowring** [comment]
@McShelby is interested in bringing new features for more than a year now.

@McShelby, probably now is the time to do a fork? I know the open maintenance issue but since the maintainers are not active nor responsive, there are not many ways left if this project should not die.

**Comment by muffmolch** [comment]
> @McShelby is interested in bringing new features for more than a year now.
> 
> @McShelby, probably now is the time to do a fork? I know the open maintenance issue but since the maintainers are not active nor responsive, there are not many ways left if this project should not die.

Yes, it seems to be the best. I don't know what's the problem in background, but he asked many times and nothing happens. Looking at his patches, I don't think it's because of bad coding ....

**Comment by lkppo** [comment]
A year off does not mean that the project is abandoned but rather that it is not a priority or that the project has exceeded the commitment that the maintainer wants to give.

Did someone offered to help him maintaining the project or become the new maintainer, if he no longer has interest?

It's much better than a fork.

**Comment by muffmolch** [comment]
> A year off does not mean that the project is abandoned but rather that it is not a priority or that the project has exceeded the commitment that the maintainer wants to give.
> 
> Did someone offered to help him maintaining the project or become the new maintainer, if he no longer has interest?
> 
> It's much better than a fork.

As @mirisbowring said: @McShelby did! Quite often, bt nothin happnes. There are also a lot of vaild PRs open ...

**Comment by muffmolch** [comment]
@lkppo And a year in software development is an immense amount of time. hugo was at v0.72 a year ago and is now at v0.84. moreover, hugo is already reporting this theme depreacted messages ... so it's only a matter of time before it drops out of the list of officially available hugo themes.

But that's the known risk you take with one man libs. There is also nothing reprehensible about it. As you say, everyone has their own priorities. And if it just does not go further, then you just have to tackle it yourself ;-)

**Comment by lkppo** [comment]
I've seen this happen several times and the maintainer has come back or passed the torch. Before forking, ask the maintainer what their position is and make an offer if there are one or two people really willing to maintain. Opening multiple PRs / Issues and waiting is not the same as politely asking the question and showing that you are ready to help. The maintainer needs to think about it in general and maybe formulate things that he can't seem to say.

If you really do not receive an answer to this kind of question then fork after a reasonable delay (1 month?).

**Comment by mirisbowring** [comment]
@McShelby offered a video conference more then a month ago to talk to the maintainers:
https://github.com/matcornic/hugo-theme-learn/issues/442

**Comment by muffmolch** [comment]
And that time, in my opinion, has simply passed. There are several threads here with offers to help support. Many small things, but also elementary things. 
From the creator's point of view, of course, it's not always easy. It's not a paid project and mostly just a byproduct that fell off somewhere else. From that point of view, I am also very grateful that this theme exists at all. But it must remain compatible to the current Hugo releases in the long run. I also use several third-party packages, some of which are out of date. There was a lot of promise here too, but in the end forking was the only viable solution, which is a shame because if it was my product I would love it to be used for a long time and would take other supporters on board. at least I would make some kind of statement. However, you have to remember that last year was an absolute exception and we don't know if that has anything to do with the discontinued fort development.

And as said now fpr many time: @McShelby would throw himself into the fire for us. Thus... let it happen :-)

**Comment by McShelby** [comment]
So at first: No offense here. No maintainer is forced to support his projects. If he has stepped away from it, so be it. The reasons for this are not mine to judge.

Nevertheless, the situation currently is still unsatisfiying for users of this theme. A lot of PRs are lingering in the queue. Some with (in my eyes) obvious bug fixes, some with more elaborate and very dedicated work. For supporters it is really discouraging if there isn't at least any feedback of a maintainer.

That's were I currently am. I already offered help in a few ways but besides from general doubt by the maintainers (which I can perfectly understand) there wasn't much of a feedback. Especially not for PRs, reviews or some perspective on how to proceed to gain more trust.

I know from personal experience, it is sometimes hard to admit a project once beloved, is now obsolete and outdated. As said above, a maintainer can decide whatever he wants to do with his projects, but in a public environment like Github it is at least fair (and in my eyes his responsiblity) to inform the community about the current state of project.

As often, communication is the key here. From that standpoint this is Schroedingers project. Not alive, not dead yet either.

**Comment by jehoshua7** [comment]
Just adding a note here so I get notifications of any activity. I've tested several hugo themes and this one was third on the search at https://github.com/search?o=desc&q=hugo+theme&s=forks&type=Repositories , plus 1.2k stars, plus I like the look of it. Are the PR's, actually pointing to real forks or just internal within this theme ?

**Comment by jehoshua7** [comment]
I installed and was testing the https://github.com/McShelby/hugo-theme-relearn/ theme. Works fine locally but if the baseURL contains a path it does not work. See https://github.com/matcornic/hugo-theme-learn/issues/25#issuecomment-873497994

@McShelby

**Comment by McShelby** [comment]
I can only speak for myself: Yes, I started a fork last week at https://github.com/McShelby/hugo-theme-relearn.

I aim to work on open issues of this repo in the 1.x.x releases and stay as compatible to this theme as possible.

The 2.x.x version may see some more significant changes, but at this moment that's more fiction than a plan.

**Comment by jehoshua7** [comment]
@McShelby Okay thanks for your reply. I uploaded your theme to a website today, and although it worked locally, it wouldn't work properly on a test website. After a few hours it suddenyl dawned on me to check the web server logs. All pages returned a 410 beccause I had Apache mods in .htaccess to look for a "?". Removed that and it worked fine. The moral of the story may be "check the web server logs first" ..lol

**Comment by McShelby** [comment]
I've forked this project to [McShelby/hugo-theme-relearn](https://github.com/McShelby/hugo-theme-relearn). The fork is meant to be a drop-in replacement of this theme.

The first release 1.0.0 was on 2021-07-01. Version 2.0.0 of my fork was released a few hours ago. All together it [fixes 57 issues](https://github.com/McShelby/hugo-theme-relearn/issues?q=is%3Aissue+is%3Aclosed), many of them are long outstanding in this theme's issue list.

Forking the theme wasn't a light decision and loomed in me for almost a year. I've made proposals on how to revive the development multiple times but with no success. Even other users are discussing the theme's state to no success (see matcornic/hugo-theme-learn#502, matcornic/hugo-theme-learn#505). I've made PRs, code reviews and giving help in the issues comments with almost no reaction from the maintainers.

Although this repo mentions the last update to be on 2021-03-31, the commit history doesn't show significant development for almost a year.

I am more than willing to dump my fork if the development of the this theme comes up to speed again. **[But for the moment the learn theme is dead](https://github.com/matcornic/hugo-theme-learn/issues/348#issuecomment-581080113)**.

**Comment by DanielRuf** [comment]
> I've forked this project to [McShelby/hugo-theme-relearn](https://github.com/McShelby/hugo-theme-relearn). The fork is meant to be a drop-in replacement of this theme.
> 
> See further statement in [#442](https://github.com/matcornic/hugo-theme-learn/issues/442#issuecomment-907863495).

Sounds good. Which of the open issues and PRs are covered by the current changes in the fork?

**Comment by McShelby** [comment]
I'ld say around 60% and still counting. You'll find references on all the fixed issues in this project.

**Comment by DanielRuf** [comment]
> [<img alt="@McShelby" width="40" height="40" src="https://avatars.githubusercontent.com/u/11670870?s=88&amp;v=4">](/McShelby)

Great. Can you need some help? I have still some issues and PRs open and maybe I could help there.

**Comment by McShelby** [comment]
Sure, you're welcome! Sadly, I've moved a lot of code, so existing PRs from this project most likely will not apply without conflicts and justification.
